source/apiVolontaria/volunteer/tests/tests_admin_Cycle.py

```diff
@@ -0,0 +1,195 @@
+            address=self.address,
+        )
+        self.cycle = Cycle.objects.create(
+            name="my cycle",
+        )
+        self.task_type = TaskType.objects.create(
+            name="my tasktype",
+        )
+
+        self.start_date = timezone.now() - timezone.timedelta(
+            minutes=100,
+        )
+        self.end_date = self.start_date + timezone.timedelta(
+            minutes=50,
+        )
+
+        self.cycle_inactive = Cycle.objects.create(
+            name="my cycle",
+            start_date=self.start_date,
+            end_date=self.end_date,
+        )
+
+        # Some date INSIDE the cycle range
+        self.start_date = self.start_date + timezone.timedelta(
+            minutes=1,
+        )
+        self.end_date = self.end_date - timezone.timedelta(
+            minutes=1,
+        )
+
+        self.event = Event.objects.create(
+            cell=self.cell,
+            cycle=self.cycle,
+            task_type=self.task_type,
+            start_date=self.start_date,
+            end_date=self.end_date,
+        )
+
+        self.event2 = Event.objects.create(
+            cell=self.cell,
+            cycle=self.cycle,
+            task_type=self.task_type,
+            start_date=self.start_date,
+            end_date=self.end_date,
+        )
+
+        self.participation = Participation(
+            event=self.event,
+            user=self.user,
+            standby=False,
+            presence_status='P'
+        )
+        self.participation.save()
+
+        self.participation2 = Participation(
+            event=self.event2,
+            user=self.user,
+            standby=False,
+            presence_status='P'
+        )
+        self.participation2.save()
+
+        self.site = AdminSite()
+        self.admin = CycleAdmin(Cycle, self.site)
+
+        self.profile = Profile.objects.filter(
+            user__pk=self.participation.user.pk
+        ).first()
+
+        self.client = Client()
+        self.client.login(username=self.username, password=self.password)
+
+    def test_admin_access(self):
+        admin_pages = [
+            "/admin/volunteer/",
+            "/admin/volunteer/cycle/",
+            ]
+
+        for page in admin_pages:
+            resp = self.client.get(page)
+            self.assertEqual(resp.status_code, 200)
+            self.assertIn("<!DOCTYPE html", str(resp.content))
+
+    def test_admin_participations(self):
+        self.assertNotEqual(self.admin, None)
+
+    def test_admin_cycle_participation_export(self):
+        change_url = urlresolvers.reverse('admin:volunteer_cycle_changelist')
+
+        data = {'action': 'generate_participation_report_csv',
+                ACTION_CHECKBOX_NAME: [self.cycle.pk, ]}
+
+        response = self.client.post(change_url, data)
+
+        self.assertEquals(response.status_code, 200)
+
+        self.assertEquals(
+            response.content,
+            b'first_name,last_name,email,total_time\r\n,,,96\r\n'
```

Comment:
indent at end of line

Reply:
Not sure about this one

Reply:
I think he meant a comma at end of line.
ie:
```
my_dict = {
    item1: "value1",
    item2: "value2",
}

Reply:
Ahah, yes i wanted to say "coma" as said @francisjeanneau :rofl: